token/program-2022-test/tests/token_group_initialize_member.rs

```diff
@@ -0,0 +1,587 @@
+        context: context.clone(),
+        token_context: None,
+    };
+
+    let mut group_test_context = create_context();
+    setup(
+        &mut group_test_context,
+        group_mint_keypair.insecure_clone(),
+        vec![ExtensionInitializationParams::GroupPointer {
+            authority: Some(group_authority.pubkey()),
+            group_address: Some(group_mint_keypair.pubkey()),
+        }],
+    )
+    .await;
+    let group_token_context = group_test_context.token_context.take().unwrap();
+
+    group_token_context
+        .token
+        .token_group_initialize_with_rent_transfer(
+            &payer.pubkey(),
+            &group_token_context.mint_authority.pubkey(),
+            &group_authority.pubkey(),
+            5,
+            &[&payer, &group_token_context.mint_authority],
+        )
+        .await
+        .unwrap();
+
+    let mut member1_test_context = create_context();
+    setup(
+        &mut member1_test_context,
+        member1_mint_keypair.insecure_clone(),
+        vec![ExtensionInitializationParams::GroupMemberPointer {
+            authority: Some(member1_authority.pubkey()),
+            member_address: Some(member1_mint_keypair.pubkey()),
+        }],
+    )
+    .await;
+    let member1_token_context = member1_test_context.token_context.take().unwrap();
+
+    let mut member2_test_context = create_context();
+    setup(
+        &mut member2_test_context,
+        member2_mint_keypair.insecure_clone(),
+        vec![ExtensionInitializationParams::GroupMemberPointer {
+            authority: Some(member2_authority.pubkey()),
+            member_address: Some(member2_mint_keypair.pubkey()),
+        }],
+    )
+    .await;
+    let member2_token_context = member2_test_context.token_context.take().unwrap();
+
+    let mut member3_test_context = create_context();
+    setup(
+        &mut member3_test_context,
+        member3_mint_keypair.insecure_clone(),
+        vec![ExtensionInitializationParams::GroupMemberPointer {
+            authority: Some(member3_authority.pubkey()),
+            member_address: Some(member3_mint_keypair.pubkey()),
+        }],
+    )
+    .await;
+    let member3_token_context = member3_test_context.token_context.take().unwrap();
+
+    // // fails without more lamports for new rent-exemption
+    // let error = member1_token_context
+    //     .token
+    //     .token_group_initialize_member(
+    //         &member1_token_context.mint_authority.pubkey(),
+    //         &group_mint_keypair.pubkey(),
+    //         &group_authority.pubkey(),
+    //         &[
+    //             &member1_token_context.mint_authority,
+    //             &group_authority,
+    //         ],
+    //     )
+    //     .await
+    //     .unwrap_err();
+    // assert_eq!(
+    //     error,
+    //     TokenClientError::Client(Box::new(TransportError::TransactionError(
+    //         TransactionError::InsufficientFundsForRent { account_index: 1 }
+    //     )))
+    // );
```

Comment:
If you re-enable this, be sure to refresh the blockhash before trying your `fail double-init` case again, since you may get the error from this transaction instead.